05_stringer/brvishnu92/main_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//main() test
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := `127.0.0.1
+`
+	actual := buf.String()
+	assert.Equalf(t, expected, actual, "Got: %v Want: %v", actual, expected)
+}
+
+func TestStringer(t *testing.T) {
+	testCases := map[string]struct {
+		input    IPAddr
+		expected string
+	}{
+		"test1":  {input: IPAddr{127, 0, 0, 1}, expected: "127.0.0.1"},
```

Comment:
I get the feeling that you have copied the same test skeleton code from your previous labs 3 and 4 into this as they all have the same three test cases (`test1, empty, oneval`).

I can see that it is beneficial to copy the skeleton code of a generic test involving multiple cases like these have been across the labs. However when constructing specific cases for the test to run you should be thinking about the different edge cases and possible outputs that might occur from inputs to this function. Rather than just assigning a generic three cases to each function, you should analyse the function and see what can happen. 

For this, what happens if you give it a value of empty, single value, two values, three values etc. And use descriptive names for test cases, `test1` is non descriptive and you won't get much out of seeing a failure message with that test name.

Can you please add more cases, and update the name of the tests accordingly.

Reply:
Done. Yeah from now on will look into the boundary cases. Thanks for that. :-)